ballerina/http_errors.bal

```diff
@@ -144,6 +144,9 @@ public type MaximumWaitTimeExceededError distinct GenericClientError;
 # Represents a cookie error that occurred when using the cookies.
 public type CookieHandlingError distinct GenericClientError;
 
+# Represents a client connector error that occurred.
+public type ClientConnectorError distinct GenericClientError;
```

Comment:
```suggestion
public type ClientConnectorError distinct (ClientError & error<Detail>);
```

Lets make it a part of `error<Detail>`, then we can access the status code through the type rather than having the `do{} on fail{}` block in the `http_connection.bal`: 
```bal
if errorResponse is ApplicationResponseError || errorResponse is ClientConnectorError {
    InternalServerError err = {
        headers: errorResponse.detail().headers,
        body: errorResponse.detail().body
    };
    response = createStatusCodeResponse(err, returnMediaType);
    response.statusCode = errorResponse.detail().statusCode;
}
```

Reply:
Fixed in [91d0caf](https://github.com/ballerina-platform/module-ballerina-http/pull/1497/commits/91d0caf1bd37f1461fed5919d2db9a5a17064fcf)